File: src/jsContext.java
import jsjava.applet.AppletContext;
import jsjava.applet.AppletStub;
import java.net.URL;

import jsjavax.swing.JApplet;


public class jsContext implements AppletStub {

	JApplet app;
	public jsContext(JApplet app) {
		this.app = app;
	}

	@Override
	public boolean isActive() {
		return true;
	}

	@Override
	public URL getDocumentBase() {
		return null;
	}

	@Override
	public URL getCodeBase() {
		return null;
	}

	@Override
	public String getParameter(String name) {
		System.out.println("get parameter: " + name);
		if ("width".equals(name))
			return "300";
		if ("height".equals(name))
			return "300";
		return null;
	}

	@Override
	public AppletContext getAppletContext() {
		return null;
	}

	@Override
	public void appletResize(int width, int height) {
		System.out.println("resize applet to " + width + " " + height);
	}

}

File: site/swingjs/j2s/jsContext.js
Clazz.load (["jsjava.applet.AppletStub"], "jsContext", null, function () {
c$ = Clazz.decorateAsClass (function () {
this.app = null;
Clazz.instantialize (this, arguments);
}, null, "jsContext", null, jsjava.applet.AppletStub);
Clazz.makeConstructor (c$, 
function (app) {
this.app = app;
}, "jsjavax.swing.JApplet");
Clazz.overrideMethod (c$, "isActive", 
function () {
return true;
});
Clazz.overrideMethod (c$, "getDocumentBase", 
function () {
return null;
});
Clazz.overrideMethod (c$, "getCodeBase", 
function () {
return null;
});
Clazz.overrideMethod (c$, "getParameter", 
function (name) {
System.out.println ("get parameter: " + name);
if ("width".equals (name)) return "300";
if ("height".equals (name)) return "300";
return null;
}, "~S");
Clazz.overrideMethod (c$, "getAppletContext", 
function () {
return null;
});
Clazz.overrideMethod (c$, "appletResize", 
function (width, height) {
System.out.println ("resize applet to " + width + " " + height);
}, "~N,~N");
});
